pkgs/state-proving-manager/src/manager.zig

```diff
@@ -1,15 +1,48 @@
-const types = @import("types")
-const state_transition = @import("state-transition")
+const std = @import("std");
+const ssz = @import("ssz");
+const types = @import("@zeam/types");
+const state_transition = @import("@zeam/state-transition");
+const allocator = std.heap.page_allocator;
 
+pub const ZKVMContext = struct {
+    program_path: []const u8,
+    output_dir: []const u8,
+    backend: ?[]const u8 = null,
+};
 
-const StateTransitionOpts = struct{
-    zk_vm: ZKVM_TYPES,
-}
+extern fn powdr_prove(serialized: [*]const u8, len: usize, output: [*]u8, output_len: usize, binary_path: [*]const u8, binary_path_length: usize, result_path: [*]const u8, result_path_len: usize) void;
+
+pub const zkvm_configs: []const ZKVMContext = &.{
+    .{ .program_path = "zig-out/bin/zeam-stf-powdr", .output_dir = "out", .backend = "plonky3" },
+};
+
+pub const StateTransitionOpts = struct {
+    zk_vm: ZKVMContext,
+};
+
+pub fn execute_transition(state: types.BeamState, block: types.SignedBeamBlock, opts: StateTransitionOpts) !types.BeamSTFProof {
+    _ = opts;
+    var prover_input = types.BeamSTFProverInput{
+        .state = state,
+        .block = block,
+    };
 
-pub fn execute_transition(state: types.BeamState, block: types.SignedBeamBlock, opts: StateTransitionOpts) types.BeamSTFProof {
+    // Verify that the execution is correct before starting the
+    // computation-intensive prover.
+    try state_transition.apply_transition(allocator, &prover_input.state, prover_input.block);
```

Comment:
this shouldn't be done as it will mutate state, anyway this `prover` will be called for a correct state transition block only